src/Tes.Runner/Docker/ContainerRegistryAuthorizationManager.cs

```diff
@@ -74,12 +79,32 @@ public static bool TryParseAzureContainerRegisteryParts(string imageName, out st
         }
 
 
-        public ContainerRegistryContentClient CreateContainerRegistryContentClientWithAcquireAuthTokenPolicy(Uri endpoint, string repositoryName, RuntimeOptions runtimeOptions, Action<string> onCapture)
+        public async Task<ContainerRegistryContentClient> CreateContainerRegistryContentClientWithAcquireAuthTokenPolicyAsync(Uri endpoint, string repositoryName, RuntimeOptions runtimeOptions, Action<string> onCapture)
         {
+            // TODO we are replacing node identity ACR auth with action identity - should do both instead?
+
+            string? terraACRIdentity = await FetchTerraACRActionIdentityAsync(runtimeOptions);
+
+
             // Use a pipeline policy to get access to the ACR access token we will need to pass to Docker.
             var clientOptions = new ContainerRegistryClientOptions();
             clientOptions.AddPolicy(new AcquireDockerAuthTokenPipelinePolicy(onCapture), HttpPipelinePosition.PerCall);
-            return new ContainerRegistryContentClient(endpoint, repositoryName, tokenCredentialsManager.GetTokenCredential(runtimeOptions), clientOptions);
+            return new ContainerRegistryContentClient(endpoint, repositoryName, tokenCredentialsManager.GetTokenCredential(runtimeOptions, null, terraACRIdentity), clientOptions);
+        }
+
+        private async Task<string?> FetchTerraACRActionIdentityAsync(RuntimeOptions runtimeOptions)
+        {
+            if (runtimeOptions.Terra is null || string.IsNullOrWhiteSpace(runtimeOptions.Terra.BillingProfileId) || string.IsNullOrWhiteSpace(runtimeOptions.Terra.SamApiHost))
+            {
+                return null;
+            }
+
+            // TODO test with invalid guid for billing profile id
```

Comment:
Adding error handling here in case either the Sam request or guid processing fail.